sdk/src/invariant.ts

```diff
@@ -414,4 +415,29 @@ export class Invariant {
       }
     )
   }
+
+  private async getPoolKeys() {
+    const key = hash('pool_keys')
+    const stateRootHash = await this.service.getStateRootHash()
+    const response = await this.service.getDictionaryItemByName(
+      stateRootHash,
+      this.contract.contractHash!,
+      'state',
+      key,
+      { rawData: true }
+    )
+
+    const rawBytes = (response.CLValue! as any).bytes
+
+    return decodePoolKeys(rawBytes)
+  }
+
+  async getPools() {
+    const poolKeys = await this.getPoolKeys()
+    const pools = []
+    for (const poolKey of poolKeys) {
+      pools.push(await this.getPool(poolKey))
```

Comment:
Could you attempt to retrieve multiple pools simultaneously? Try to use `Promise.all`